daemon/daemon_test.go

```diff
@@ -335,6 +337,54 @@ func TestDaemonReloadLabels(t *testing.T) {
 	}
 }
 
+func TestDaemonReloadInsecureRegistries(t *testing.T) {
+	daemon := &Daemon{}
+	daemon.RegistryService = registry.NewService(registry.ServiceOptions{})
+	daemon.configStore = &Config{}
```

Comment:
Seems like we should have a test with an already existing insecure registry that is not in the new config (which I would expect to be gone when the new config is loaded).